PMSCore/ReportManager.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using PMSCore;
+
+namespace PMSCore
+{
+    /// <summary>
+    /// This class handles the creation and management of reports.
+    /// </summary>
+    public sealed class ReportManager
+    {
+        // The single instance of ReportManager
+        private static readonly ReportManager _instance = new ReportManager();
+
+        /// <summary>
+        /// Private constructor to prevent external instantiation.
+        /// </summary>
+        private ReportManager()
+        {
+        }
+
+        /// <summary>
+        /// Gets the single instance of ReportManager.
+        /// </summary>
+        public static ReportManager Instance => _instance;
```

Comment:
This should be private and controlled with a GetInstance() method like TaskManager and ProjectManager() and deletable using a ResetInstance()

Reply:
@DistroRickDev , this should be done now, can you please double-check?